### PR TITLE
Return the local variable directly, make RVO enabled

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
/kind improvement

### CODE REVIEW

File: internal/core/src/segcore/SegmentSealedImpl.cpp [hunk 1]
```diff
@@ -582,7 +582,7 @@ SegmentSealedImpl::search_ids(const BitsetView& bitset, Timestamp timestamp) con
             dst_offset.emplace_back(SegOffset(i));
         }
     }
-    return std::move(dst_offset);
+    return dst_offset;
```
Comment (soothing-rain) [inline internal/core/src/segcore/SegmentSealedImpl.cpp:585]: Side question, shall we remove the `std::move` call here as well? https://github.com/milvus-io/milvus/blob/804fbb6e27a56c0cacf8f6b4d0b2e173e3c4fe7e/internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp#L134

Comment (yah01) [reply internal/core/src/segcore/SegmentSealedImpl.cpp:585]: Yes, I will remove it.

**Pull request merged**